dotnet/src/SemanticKernel/Planning/InstrumentedPlan.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// Standard Semantic Kernel callable plan with instrumentation.
+/// </summary>
+public sealed class InstrumentedPlan : IPlan
+{
+    /// <inheritdoc/>
+    public string Name => this._plan.Name;
+
+    /// <inheritdoc/>
+    public string SkillName => this._plan.SkillName;
+
+    /// <inheritdoc/>
+    public string Description => this._plan.Description;
+
+    /// <inheritdoc/>
+    public bool IsSemantic => this._plan.IsSemantic;
+
+    /// <inheritdoc/>
+    public CompleteRequestSettings RequestSettings => this._plan.RequestSettings;
+
+    /// <summary>
+    /// Initialize a new instance of the <see cref="InstrumentedPlan"/> class.
+    /// </summary>
+    /// <param name="plan">Instance of <see cref="IPlan"/> to decorate.</param>
+    /// <param name="logger">Optional logger.</param>
+    public InstrumentedPlan(
+        IPlan plan,
+        ILogger? logger = null)
+    {
+        this._plan = plan;
+        this._logger = logger ?? NullLogger.Instance;
+    }
+
+    /// <inheritdoc/>
+    public FunctionView Describe()
+    {
+        return this._plan.Describe();
+    }
+
+    /// <inheritdoc/>
+    public async Task<SKContext> InvokeAsync(
+        SKContext context,
+        CompleteRequestSettings? settings = null,
+        CancellationToken cancellationToken = default)
+    {
+        return await this.InvokeWithInstrumentationAsync(() =>
+            this._plan.InvokeAsync(context, settings, cancellationToken)).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task<SKContext> InvokeAsync(
+        string? input = null,
+        CompleteRequestSettings? settings = null,
+        ILogger? logger = null,
+        CancellationToken cancellationToken = default)
+    {
+        return await this.InvokeWithInstrumentationAsync(() =>
+            this._plan.InvokeAsync(input, settings, logger ?? this._logger, cancellationToken)).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public ISKFunction SetAIConfiguration(CompleteRequestSettings settings)
+    {
+        return this._plan.SetAIConfiguration(settings);
+    }
```

Comment:
Just a thought: for these 1-line proxy functions, using the inlined syntax"

`public ISKFunction SetAIConfiguration(CompleteRequestSettings settings) => this._plan.SetAIConfiguration(settings);`

...might be a good idea. It produces shorter files for one thing, but it also means there's less opportunity for error (you can only do one thing and it implicitly returns.

Optional - I'll leave it to you.

Reply:
Good idea, updated!